ui/src/layouts/chat.rs

```diff
@@ -37,3 +108,91 @@ pub fn ChatLayout(cx: Scope) -> Element {
         }
     ))
 }
+
+async fn drop_and_attach_files(
+    eval: UseEvalFn,
+    window: &DesktopContext,
+    drag_event: &UseRef<Option<FileDropEvent>>,
+    state: UseSharedState<State>,
+) {
+    let new_files = drag_and_drop_function(eval.clone(), window, drag_event).await;
+    let mut new_files_to_upload: Vec<_> = state
+        .read()
+        .get_active_chat()
+        .map(|f| f.files_attached_to_send)
+        .unwrap_or_default()
+        .iter()
+        .filter(|file_name| !new_files.contains(file_name))
+        .cloned()
+        .collect();
+    new_files_to_upload.extend(new_files);
+    let chat_uuid = state
+        .read()
+        .get_active_chat()
+        .map(|f| f.id)
+        .unwrap_or(Uuid::nil());
+    state
+        .write()
+        .mutate(Action::SetChatAttachments(chat_uuid, new_files_to_upload));
+}
+
+// Like ui::src:layout::storage::drag_and_drop_function
+async fn drag_and_drop_function(
+    eval: UseEvalFn,
+    window: &DesktopContext,
+    drag_event: &UseRef<Option<FileDropEvent>>,
+) -> Vec<PathBuf> {
+    *drag_event.write_silent() = Some(get_drag_event::get_drag_event());
+    let mut new_files_to_upload = Vec::new();
+    loop {
+        let file_drop_event = get_drag_event::get_drag_event();
+        match file_drop_event {
+            FileDropEvent::Hovered { paths, .. } => {
+                if verify_if_are_valid_paths(&paths) {
+                    let mut script = OVERLAY_SCRIPT.replace("$IS_DRAGGING", "true");
+                    if paths.len() > 1 {
+                        script.push_str(&FEEDBACK_TEXT_SCRIPT.replace(
+                            "$TEXT",
+                            &format!(
+                                "{} {}!",
+                                paths.len(),
+                                get_local_text("files.files-to-upload")
+                            ),
+                        ));
+                    } else {
+                        script.push_str(&FEEDBACK_TEXT_SCRIPT.replace(
+                            "$TEXT",
+                            &format!(
+                                "{} {}!",
+                                paths.len(),
+                                get_local_text("files.one-file-to-upload")
+                            ),
+                        ));
+                    }
```

Comment:
also don't need to fix this now but it could be condensed to
```
let local_text = if paths.len() > 1 { ... } else { ... };
script.push_str(...);
```